### PR TITLE
Adding the differential privacy to the CountAccumulator.

### PR DESCRIPTION
**Pull request by lovroprepolec**
Need to figure out how to obtain eps and delta from the BudgetAccountant. Tests with noise will be added after the parameters eps and delta can be set from BudgetAccountant.

## Description
Adding the differential privacy to the CountAccumulator.

## Affected Dependencies
none.

## How has this been tested?
Unit tests.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: pipeline_dp/accumulator.py [hunk 1]
```diff
@@ -29,10 +29,21 @@ def create_accumulator_params(
...
     accumulator_params = []
     if pipeline_dp.Metrics.COUNT in aggregation_params.metrics:
-        # TODO: populate CountParams from budget_accountant when it is ready
+        request_budget = budget_accountant.request_budget
+        low, high, max_partitions_contributed,\
+        max_contributions_per_partition = \
+            aggregation_params.low, aggregation_params.high, \
+            aggregation_params.max_partitions_contributed, \
+            aggregation_params.max_contributions_per_partition
+        noise_kind = NoiseKind.GAUSSIAN
```
Comment (dvadym) [inline pipeline_dp/accumulator.py:38]: Ah, I missed to add noise_kind to aggregation_params. Could you please add `noise_kind` to aggregation_params ?

File: pipeline_dp/accumulator.py [hunk 2]
```diff
@@ -190,8 +202,8 @@ def add_accumulator(self,
         return self
 
     def compute_metrics(self) -> float:
-        # TODO: add differential privacy
-        return self._count
+        return pipeline_dp.dp_computations.compute_dp_count(
```
Comment (dvadym) [inline pipeline_dp/accumulator.py:205]: Nit: no need in `pipeline_dp`, `dp_computations` is already imported

File: tests/accumulator_test.py [hunk 3]
```diff
@@ -293,23 +293,24 @@ def compute_metrics(self):
 class CountAccumulatorTest(unittest.TestCase):
 
     def test_without_noise(self):
+        no_noise = (1, 1, 0, 0, 0, 0, pipeline_dp.NoiseKind.GAUSSIAN)
```
Comment (dvadym) [inline tests/accumulator_test.py:296]: Please add a test with noise. It's enough to have one test case. 
Here is an [example](https://github.com/OpenMined/PipelineDP/blob/1505b288b620b019080eca5c7b90915633bc45cd/tests/accumulator_test.py#L342) how to have test with noise

**Review by dvadym — COMMENTED** [review]
Thanks, it looks good. I left some comments

File: pipeline_dp/accumulator.py (hunk 1 above)
Comment (dvadym) [reply pipeline_dp/accumulator.py:38]: noise_kind was added to aggregate_params on the PR merged today

File: tests/accumulator_test.py (hunk 3 above)
Comment (lovroprepolec) [reply tests/accumulator_test.py]: Done.

**Review by dvadym — COMMENTED** [review]
Thanks, it looks good. A few minor comments

File: pipeline_dp/accumulator.py [hunk 4]
```diff
@@ -194,15 +195,52 @@ def create(self, values: typing.List) -> Accumulator:
         return CompoundAccumulator(accumulators)
 
 
-@dataclass
-class CountParams:
+class AggregatorParams:
```
[the remark this answers is not in the document]
Comment (lovroprepolec) [reply pipeline_dp/accumulator.py:198]: AccumulatorParams exists already. I have named it AccumulatorClassParams.

**Review by dvadym — APPROVED** [review]
Thank you for implementing DP count!

**Pull request merged**